src/Kovan-DssSpell.t.sol

```diff
@@ -1278,71 +1698,390 @@ contract DssSpellTest is DSTest, DSMath {
+            bytes32(uint256(1))
+        );
+        assertEq(end.wards(address(this)), 1);
 
-            hevm.warp(monday_1400_UTC - 3 days);                       // Friday,   14:00 UTC
-            assertEq(spell.nextCastTime(), monday_1400_UTC - 3 days);  // Able to cast
+        end.cage();
+        end.cage("ETH-A");
+        end.cage("BAT-A");
+        end.cage("LINK-A");
 
-            hevm.warp(monday_2100_UTC);                                // Monday,   21:00 UTC
-            assertEq(spell.nextCastTime(), monday_1400_UTC + 1 days);  // Tuesday,  14:00 UTC
+        (,, address usr,,,,,) = flipETHA.bids(auctionIdETHA);
+        assertTrue(usr != address(0));
+        (,, usr,,,,,) = flipBATA.bids(auctionIdBATA);
+        assertTrue(usr != address(0));
+        (,,, usr,,) = clipLINKA.sales(auctionIdLINKA);
+        assertTrue(usr != address(0));
 
-            hevm.warp(monday_2100_UTC - 1 days);                       // Sunday,   21:00 UTC
-            assertEq(spell.nextCastTime(), monday_1400_UTC);           // Monday,   14:00 UTC
+        end.skip("ETH-A", auctionIdETHA);
+        end.skip("BAT-A", auctionIdBATA);
+        end.snip("LINK-A", auctionIdLINKA);
 
-            hevm.warp(monday_2100_UTC - 2 days);                       // Saturday, 21:00 UTC
-            assertEq(spell.nextCastTime(), monday_1400_UTC);           // Monday,   14:00 UTC
+        (,, usr,,,,,) = flipETHA.bids(auctionIdETHA);
+        assertTrue(usr == address(0));
+        (,, usr,,,,,) = flipBATA.bids(auctionIdBATA);
+        assertTrue(usr == address(0));
+        (,,, usr,,) = clipLINKA.sales(auctionIdLINKA);
+        assertTrue(usr == address(0));
 
-            hevm.warp(monday_2100_UTC - 3 days);                       // Friday,   21:00 UTC
-            assertEq(spell.nextCastTime(), monday_1400_UTC);           // Monday,   14:00 UTC
+        end.skim("ETH-A", address(this));
+        end.skim("BAT-A", address(this));
+        end.skim("LINK-A", address(this));
 
-            // Time tests
-            uint256 castTime;
+        end.free("ETH-A");
+        end.free("BAT-A");
+        end.free("LINK-A");
 
-            for(uint256 i = 0; i < 5; i++) {
-                castTime = monday_1400_UTC + i * 1 days; // Next day at 14:00 UTC
-                hevm.warp(castTime - 1 seconds); // 13:59:59 UTC
-                assertEq(spell.nextCastTime(), castTime);
+        hevm.warp(block.timestamp + end.wait());
 
-                hevm.warp(castTime + 7 hours + 1 seconds); // 21:00:01 UTC
-                if (i < 4) {
-                    assertEq(spell.nextCastTime(), monday_1400_UTC + (i + 1) * 1 days); // Next day at 14:00 UTC
-                } else {
-                    assertEq(spell.nextCastTime(), monday_1400_UTC + 7 days); // Next monday at 14:00 UTC (friday case)
-                }
-            }
-        }
+        vow.heal(min(vat.dai(address(vow)), sub(sub(vat.sin(address(vow)), vow.Sin()), vow.Ash())));
+
+        end.thaw();
```

Comment:
It's worth noting that `end.thaw()` requires `vat.dai(address(vow)) == 0`, the `min` is unnecessary...either you can `heal` away the Vow's DAI balance, or `thaw` will revert.

Reply:
Good one.